mlrun/datastore/datastore_profile.py

```diff
@@ -198,6 +203,14 @@ def attributes(self) -> dict[str, typing.Any]:
         return attributes
 
 
+@deprecated(
+    "This class is deprecated from mlrun 1.10.0, and will be removed in 1.12.0. "
+    "Use `DatastoreProfileKafkaStream` instead."
+)
+class DatastoreProfileKafkaSource(DatastoreProfileKafkaStream):
+    type: str = pydantic.v1.Field("kafka_source")
```

Comment:
In order to keep bc, I suggest following change: 
- DatastoreProfileKafkaSource will not override type kafka_source and will create kafka_stream
- in _DATASTORE_TYPE_TO_PROFILE_CLASS , change  "kafka_source": DatastoreProfileKafkaSource, to  "kafka_source": DatastoreProfileKafkaStream
This way all the old profiles created as kafka_source will generate kafka_stream objects, and all the DatastoreProfileKafkaSource objects will be saved as DatastoreProfileKafkaStream 

Also add "TODO: Remove in 1.12.0" in all the places that should be removed in 1.12